code/modules/atmospherics/machinery/airalarm.dm

```diff
@@ -329,35 +330,35 @@
 			var/datum/gas_mixture/gas = location.remove_air(0.25 * environment.total_moles())
 			if(!gas)
 				return
-			if(!regulating_temperature)
-				regulating_temperature = 1
+			if(!regulating_temperature && thermostat_state == TRUE)
+				regulating_temperature = TRUE
 				visible_message("\The [src] clicks as it starts [environment.temperature > target_temperature ? "cooling" : "heating"] the room.", "You hear a click and a faint electronic hum.")
 
 			if(target_temperature > MAX_TEMPERATURE)
 				target_temperature = MAX_TEMPERATURE
 
 			if(target_temperature < MIN_TEMPERATURE)
 				target_temperature = MIN_TEMPERATURE
+			if(thermostat_state == TRUE)
+				var/heat_capacity = gas.heat_capacity()
+				var/energy_used = max(abs(heat_capacity * (gas.temperature - target_temperature) ), MAX_ENERGY_CHANGE)
```

Comment:
Nitpick, but could you add a space between these lines to help with readability?
```suggestion
			if(target_temperature < MIN_TEMPERATURE)
				target_temperature = MIN_TEMPERATURE

			if(thermostat_state == TRUE)
				var/heat_capacity = gas.heat_capacity()
				var/energy_used = max(abs(heat_capacity * (gas.temperature - target_temperature) ), MAX_ENERGY_CHANGE)
```